File: platform-mobile/src/main/java/ru/pocketbyte/locolaser/platform/mobile/resource/file/IosPlistResourceFile.java
package ru.pocketbyte.locolaser.platform.mobile.resource.file;

import java.io.File;
import java.io.IOException;

public class IosPlistResourceFile extends AbsIosStringsResourceFile {

    public IosPlistResourceFile(File file, String locale) {
        super(file, locale);
    }

    @Override
    protected String getKeyValueLinePattern() {
        return "[{26}]((?:[^\"]||\\\\\")+?)\\s*=\\s*\"((?:[^\"]|\\\\\")*)\"\\s*;";
    }

    @Override
    protected void writeKeyValueString(String key, String value) throws IOException {
        writeString(key);
        writeString(" = \"");
        writeString(value);
        writeString("\";");
    }
}

File: platform-mobile/src/main/java/ru/pocketbyte/locolaser/platform/mobile/resource/file/IosPlistResourceFile.kt
package ru.pocketbyte.locolaser.platform.mobile.resource.file

import java.io.File
import java.io.IOException

class IosPlistResourceFile(file: File, locale: String) : AbsIosStringsResourceFile(file, locale) {

    override val keyValueLinePattern = "^\\s?((?:[^\"]|\\\\\")+?)\\s*=\\s*\"((?:[^\"]|\\\\\")*)\"\\s*;"

    @Throws(IOException::class)
    override fun writeKeyValueString(key: String, value: String) {
        writeString(key)
        writeString(" = \"")
        writeString(value)
        writeString("\";")
    }
}